services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/flows/FlowCrudSpec.groovy

```diff
@@ -280,99 +280,37 @@ class FlowCrudSpec extends BaseSpecification {
-                }
-            ],
-            [
-                conflict: "vlan vs no vlan on same port on dst",
-                makeFlowsConflicting: { FlowPayload dominantFlow, FlowPayload flowToConflict ->
-                    flowToConflict.destination.portNumber = dominantFlow.destination.portNumber
-                    dominantFlow.destination.vlanId = 0
-                },
-                getError: { FlowPayload flowToError ->
-                    portError(flowToError.destination.portNumber, flowToError.destination.datapath, flowToError.id)
-                }
-            ],
-            [
-                conflict: "vlan vs no vlan on same port on src",
-                makeFlowsConflicting: { FlowPayload dominantFlow, FlowPayload flowToConflict ->
-                    flowToConflict.source.portNumber = dominantFlow.source.portNumber
-                    dominantFlow.source.vlanId = 0
-                },
-                getError: { FlowPayload flowToError ->
-                    portError(flowToError.source.portNumber, flowToError.source.datapath, flowToError.id)
-                }
-            ],
-            [
-                conflict: "no vlans both flows on same port on src",
-                makeFlowsConflicting: { FlowPayload dominantFlow, FlowPayload flowToConflict ->
-                    flowToConflict.source.portNumber = dominantFlow.source.portNumber
-                    flowToConflict.source.vlanId = 0
-                    dominantFlow.source.vlanId = 0
-                },
-                getError: { FlowPayload flowToError ->
-                    portError(flowToError.source.portNumber, flowToError.source.datapath, flowToError.id)
-                }
-            ],
-            [
-                conflict: "no vlans both flows on same port on dst",
-                makeFlowsConflicting: { FlowPayload dominantFlow, FlowPayload flowToConflict ->
-                    flowToConflict.destination.portNumber = dominantFlow.destination.portNumber
-                    flowToConflict.destination.vlanId = 0
-                    dominantFlow.destination.vlanId = 0
-                },
-                getError: { FlowPayload flowToError ->
-                    portError(flowToError.destination.portNumber, flowToError.destination.datapath, flowToError.id)
-                }
-            ],
-            [
-                conflict: "the same flowId",
-                makeFlowsConflicting: { FlowPayload dominantFlow, FlowPayload flowToConflict ->
-                    flowToConflict.id = dominantFlow.id
-                },
-                getError: { FlowPayload flowToError ->
-                    "Could not create flow: Flow $flowToError.id already exists"
-                }
-            ]
-        ]
+        where:
+        data << getConflictingData()
+    }
+
+    @Unroll("Unable to update flow with #data.conflict")
+    def "Unable to update flow with conflicting vlans or flow IDs"() {
```

Comment:
During update there is also a possibility to have initial 2 flows on different switches and then update the src or dst _switch_ so that they conflict. Does it make sense?

Reply:
Probably yes, I am not sure. Anyway, I think it's better to do with another patch.